robottelo/hosts.py

```diff
@@ -249,21 +248,53 @@ def install_katello_host_tools(self):
         if result.status != 0:
             raise ContentHostError('Failed to install katello-host-tools')
 
-    def install_katello_ca(self, sat_hostname=None):
-        """Downloads and installs katello-ca rpm on the broker virtual machine.
+    def install_katello_ca(self, satellite):
+        """Downloads and installs katello-ca rpm on the content host.
 
-        Uses common helper `install_katello_ca(hostname=None)`, but passes
-        `self.hostname` instead of the hostname as we are using fake hostnames
-        for broker virtual machines.
+        :param str satellite: robottelo.hosts.Satellite instance
 
         :return: None.
         :raises robottelo.hosts.ContentHostError: If katello-ca wasn't
             installed.
         """
-        try:
-            install_katello_ca(hostname=self.hostname, sat_hostname=sat_hostname)
-        except AssertionError:
-            raise ContentHostError('Failed to download and install the katello-ca rpm')
+        cert_rpm_url = satellite.url_katello_ca_rpm
+        self.execute(f'rpm -Uvh {cert_rpm_url}')
+        # Not checking the return_code here, as rpm could be installed before
+        # and installation may fail
+        result = self.execute(f'rpm -q katello-ca-consumer-{satellite.hostname}')
+        # Checking the return_code here to verify katello-ca rpm is actually
+        # present in the system
+        if result.return_code != 0:
+            ContentHostError('Failed to download and install the katello-ca rpm')
+
+    def remove_katello_ca(self):
+        """Removes katello-ca rpm from the broker virtual machine.
+
+        :param: str capsule: (optional) Capsule hostname
+        :return: None.
+        :raises robottelo.hosts.ContentHostError: If katello-ca wasn't removed.
+        """
+        # Not checking the return_code here, as rpm can be not even installed
+        # and deleting may fail
+        self.execute('yum erase -y $(rpm -qa |grep katello-ca-consumer)')
+        # Checking the return_code here to verify katello-ca rpm is actually
+        # not present in the system
+        result = self.execute('rpm -qa |grep katello-ca-consumer')
+        if result.return_code == 0:
+            raise ContentHostError(f'katello-ca rpm(s) are still installed: {result.stdout}')
+        # Resetting rhsm.conf to point to cdn
+        # TODO don't hardcode these endpoints,
+        # read them from settings.repos or robottelo.constants.repos
+        rhsm_updates = [
```

Comment:
We should be able to just use `subscription-manager clean` to reset

Reply:
updating, can replace this whole block with `self.execute`